tools/SampleAnalyzer/Process/Plot/Histo.h

```diff
@@ -35,187 +33,142 @@
+        //                        data members
+        // -------------------------------------------------------------
+    protected:
+        /// Each variable is defined with WEIGHTS object which includes positive and negative accessors
+        /// these are for positive and negative bins. std::map<MAint32,WEIGHTS> contains a map of
+        /// different PDF and their corresponding positive and negative weights.
+
+        /// Histogram arrays
+        std::vector<std::map<MAint32, WEIGHTS>> histo_;
+        std::map<MAint32, WEIGHTS> underflow_;
+        std::map<MAint32, WEIGHTS> overflow_;
+
+        /// Histogram description
+        MAuint32 nbins_;
+        MAfloat64 xmin_;
+        MAfloat64 xmax_;
+        MAfloat64 step_;
+
+        /// Sum of event-weights over entries
+        std::map<MAint32, WEIGHTS> sum_w_;
+
+        /// Sum of squared weights
+        std::map<MAint32, WEIGHTS> sum_ww_;
+
+        /// Sum of value * weight
+        std::map<MAint32, WEIGHTS> sum_xw_;
+
+        /// Sum of value * value * weight
+        std::map<MAint32, WEIGHTS> sum_xxw_;
+
+        /// RegionSelections attached to the histo
+        std::vector<RegionSelection *> regions_;
+
+        // -------------------------------------------------------------
+        //                       method members
+        // -------------------------------------------------------------
+    public:
+        /// Constructor without argument
+        Histo() : PlotBase()
+        {
+            nbins_ = 100;
+            xmin_ = 0;
+            xmax_ = 100;
+            step_ = (xmax_ - xmin_) / static_cast<MAfloat64>(nbins_);
+        }
+
+        /// Constructor with argument
+        Histo(const std::string &name) : PlotBase(name) {}
+
+        /// Constructor with argument
+        Histo(const std::string &name, MAuint32 nbins, MAfloat64 xmin, MAfloat64 xmax) : PlotBase(name)
+        {
+            // Setting the description: nbins
+            try
+            {
+                if (nbins == 0)
+                    throw EXCEPTION_WARNING("nbins cannot be equal to 0. Set 100.", "", 0);
+                nbins_ = nbins;
+            }
+            catch (const std::exception &e)
+            {
+                MANAGE_EXCEPTION(e);
+                nbins_ = 100;
+            }
+
+            // Setting the description: min & max
+            try
+            {
+                if (xmin >= xmax)
+                    throw EXCEPTION_WARNING("xmin cannot be equal to or greater than xmax. Setting xmin to 0 and xmax to 100.", "", 0);
+                xmin_ = xmin;
+                xmax_ = xmax;
+            }
+            catch (const std::exception &e)
+            {
+                MANAGE_EXCEPTION(e);
+                xmin_ = 0.;
+                xmax_ = 100.;
+            }
+
+            step_ = (xmax_ - xmin_) / static_cast<MAfloat64>(nbins_);
+
+            // histo_.resize(nbins_);
```

Comment:
I guess this is where the `_initialize` method is useful. But we should not forget to include it at the Python levle.

Reply:
The problem is, at this stage I dont know which weights are there so i can not initialise. I will learn about the weights when the analysis is executed.

Reply:
```suggestion
            histo_.resize(nbins_);
```